cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -66,6 +66,10 @@ type InitConfiguration struct {
 	// Patches contains options related to applying patches to components deployed by kubeadm during
 	// "kubeadm init".
 	Patches *Patches
+
+	// Timeout holds timeout fields that relate to registering a new control-plane or node to the cluster,
+	// either via "kubeadm init" or "kubeadm join"
```

Comment:
```suggestion
	// Timeouts holds fields controlling timeouts when registering a control plane node during
	// "kubeadm init".
```

similar for other places,

replace "init" -> "join" for JoinTimeouts.